Feat/menu

181108-cherry opened the pull request:
## 📌 관련 이슈
- Closes #36



## ✨ 기능 요약

메뉴 엔티티, 레포지토리 추가

181108-cherry committed:
first commit

181108-cherry committed:
feat: 메뉴 엔티티, 레포지토리 추가 #36

181108-cherry committed:
로컬 변경사항 백업 커밋

181108-cherry committed:
feat: 메뉴 엔티티, 레포지토리 추가 #36

dnjs5024 committed:
Delete http/comment.http

dnjs5024 committed:
Delete src/main/java/com/example/delivery/menu/controller/MenuController.java

dnjs5024 committed:
Delete src/main/java/com/example/delivery/menu/dto/MenuRequestDto.java

dnjs5024 committed:
Delete src/main/java/com/example/delivery/menu/dto/MenuResponseDto.java

dnjs5024 committed:
Delete src/main/java/com/example/delivery/menu/service/MenuService.java

dnjs5024 committed:
Delete src/main/java/com/example/delivery/review/entity/Review.java

dnjs5024 reviewed (approved):
good

seseseseo commented on src/main/java/com/example/delivery/menu/entity/Menu.java:
```diff
@@ -0,0 +1,58 @@
+@Table(name = "menus")
+@Getter
+@SQLDelete(sql = "update menus SET is_deleted = true WHERE id = ?")
+@Where(clause = "is_deleted = false")
```
오 이게 데이터를 직접 삭제하지 않고 숨기는 방식으로 처리한 건가요? 
논리 삭제가 이런거군요 하나 배우고 갑니다!

seseseseo commented on src/main/java/com/example/delivery/menu/entity/Menu.java:
```diff
@@ -0,0 +1,58 @@
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long Id;
```
Id가 대문자인데 id로 바꿔주시는게 좋을 거 같아요!!

seseseseo commented on src/main/java/com/example/delivery/menu/entity/Menu.java:
```diff
@@ -0,0 +1,58 @@
+    private String description;
+
+    @Column(nullable = false)
+    private boolean isDeleted = false;
```
이부분은 제가 설명을 드리고 싶은데.. 잘못하겠어서 참고 블로그 링크 드립니다!
https://medium.com/@baejae/boolean%EC%97%90-is-%EB%B6%99%EC%9D%B4%EC%A7%80-%EB%A7%88%EC%84%B8%EC%9A%94-7b717246d942

https://jungguji.github.io/2020/12/31/RequestBody-Annotation-%EC%82%AC%EC%9A%A9-%EC%8B%9C-boolean-%EB%B3%80%EC%88%98-%EB%B0%94%EC%9D%B8%EB%94%A9-%EC%97%90%EB%9F%AC/

seseseseo commented on src/main/java/com/example/delivery/menu/repository/MenuRepository.java:
```diff
@@ -0,0 +1,18 @@
+                () -> new NotFoundException(ErrorCode.MENU_NOT_FOUND)
+        );
+    }
+
```
findByIdOrElseThrow() 메서드를 만든 점 좋은 방법인 거 같습니다
하나 배우고 갑니다!
엔티티 작성하시느라 고생하셨어요

181108-cherry committed:
refac: Id에서 id로 수정 및 boolean 타입 is 생략

181108-cherry committed:
refac: Id에서 id로 수정 및 boolean 타입 is 생략

# Conflicts:
#	src/main/java/com/example/delivery/menu/service/MenuService.java
